### PR TITLE
Feature/backup compressor/remove meta data

### PR DESCRIPTION
**Pull request by kpango**
### Description:
v1 new design API duplicates metadata field backup functionality.
In this PR removes thease duplicated field from compress/backup/internal code.

### Environment:

- Go Version: 1.15.2
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.1

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [x] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your core changes, as applicable?
- [ ] Have you successfully ran tests with your changes locally?

### Checklist:

- [ ] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [ ] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [ ] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/db/rdb/mysql/mysql.go [hunk 1]
```diff
@@ -175,41 +174,41 @@ func (m *mySQLClient) Close(ctx context.Context) error {
 	return nil
 }
 
-// GetMeta gets the metadata and podIPs which have index of metadata's vector.
-func (m *mySQLClient) GetMeta(ctx context.Context, uuid string) (MetaVector, error) {
+// GetVector gets the vecor adata and podIPs which have index of vector data's vector.
```
Comment (kmrmt) [inline internal/db/rdb/mysql/mysql.go:177]: typo `vecor adata`

File: pkg/manager/backup/mysql/handler/grpc/handler.go [hunk 2]
```diff
@@ -96,63 +96,63 @@ func (s *server) Locations(ctx context.Context, req *payload.Backup_Locations_Re
...
-	m, err := toModelMetaVector(meta)
+	uuid := vector.GetUuid()
+	m, err := toModelVector(vector)
 	if err != nil {
 		log.Errorf("[Register]\tunknown error\t%+v", err)
 		if span != nil {
 			span.SetStatus(trace.StatusCodeInternal(err.Error()))
 		}
-		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's could not convert vector to meta_vector", uuid), err, info.Get())
+		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's could not convert vector to vector_vector", uuid), err, info.Get())
```
Comment (kmrmt) [inline pkg/manager/backup/mysql/handler/grpc/handler.go:113]: same reason(strange error message).

File: pkg/manager/backup/cassandra/handler/grpc/handler.go [hunk 3]
```diff
@@ -105,63 +105,63 @@ func (s *server) Locations(ctx context.Context, req *payload.Backup_Locations_Re
...
-	m, err := toModelMetaVector(meta)
+	uuid := vector.GetUuid()
+	m, err := toModelVector(vector)
 	if err != nil {
 		log.Errorf("[Register]\tunknown error\t%+v", err)
 		if span != nil {
 			span.SetStatus(trace.StatusCodeInternal(err.Error()))
 		}
-		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's could not convert vector to meta_vector", uuid), err, info.Get())
+		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's could not convert vector to vector_vector", uuid), err, info.Get())
```
Comment (kmrmt) [inline pkg/manager/backup/cassandra/handler/grpc/handler.go:122]: The error message is strange.

File: pkg/manager/backup/cassandra/handler/grpc/handler.go [hunk 4]
```diff
@@ -105,63 +105,63 @@ func (s *server) Locations(ctx context.Context, req *payload.Backup_Locations_Re
...
 
-	err = s.cassandra.SetMeta(ctx, m)
+	err = s.cassandra.SetVector(ctx, m)
 	if err != nil {
 		log.Errorf("[Register]\tunknown error\t%+v", err)
 		if span != nil {
 			span.SetStatus(trace.StatusCodeInternal(err.Error()))
 		}
-		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's failed to backup metadata", uuid), err, info.Get())
+		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's failed to backup vectordata", uuid), err, info.Get())
```
Comment (kmrmt) [inline pkg/manager/backup/cassandra/handler/grpc/handler.go:131]: I think data is unnecessary.

File: pkg/manager/backup/cassandra/handler/grpc/handler.go [hunk 5]
```diff
@@ -105,63 +105,63 @@ func (s *server) Locations(ctx context.Context, req *payload.Backup_Locations_Re
...
+	for _, vector := range vectors.Vectors {
+		var m *model.Vector
+		m, err = toModelVector(vector)
 		if err != nil {
 			log.Errorf("[RegisterMulti]\tunknown error\t%+v", err)
 			if span != nil {
 				span.SetStatus(trace.StatusCodeInternal(err.Error()))
 			}
-			return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API uuids %s's could not convert vector to meta_vector", meta.GetUuid()), err, info.Get())
+			return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API uuids %s's could not convert vector to vector_vector", vector.GetUuid()), err, info.Get())
```
Comment (kmrmt) [inline pkg/manager/backup/cassandra/handler/grpc/handler.go:153]: same reason(strange error message).

File: pkg/manager/backup/cassandra/handler/grpc/handler.go [hunk 6]
```diff
@@ -105,63 +105,63 @@ func (s *server) Locations(ctx context.Context, req *payload.Backup_Locations_Re
...
 
-	err = s.cassandra.SetMetas(ctx, ms...)
+	err = s.cassandra.SetVectors(ctx, ms...)
 	if err != nil {
 		log.Errorf("[RegisterMulti]\tunknown error\t%+v", err)
 		if span != nil {
 			span.SetStatus(trace.StatusCodeInternal(err.Error()))
 		}
-		return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API failed to backup metadatas %#v", ms), err, info.Get())
+		return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API failed to backup vectordatas %#v", ms), err, info.Get())
```
Comment (kmrmt) [inline pkg/manager/backup/cassandra/handler/grpc/handler.go:164]: same reason(data is unnecessary)

File: internal/db/rdb/mysql/mysql.go (hunk 1 above)
Comment (rinx) [inline internal/db/rdb/mysql/mysql.go:177]:
```suggestion
// GetVector gets the vector data and podIPs which have index of vector data's vector.
```

File: pkg/manager/backup/mysql/handler/grpc/handler.go [hunk 7]
```diff
@@ -96,63 +96,63 @@ func (s *server) Locations(ctx context.Context, req *payload.Backup_Locations_Re
...
 
-	err = s.mysql.SetMeta(ctx, m)
+	err = s.mysql.SetVector(ctx, m)
 	if err != nil {
 		log.Errorf("[Register]\tunknown error\t%+v", err)
 		if span != nil {
 			span.SetStatus(trace.StatusCodeInternal(err.Error()))
 		}
-		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's failed to backup metadata", uuid), err, info.Get())
+		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's failed to backup vectordata", uuid), err, info.Get())
```
Comment (kmrmt) [inline pkg/manager/backup/mysql/handler/grpc/handler.go:122]: same reason(data is unnecessay).

File: pkg/manager/backup/mysql/handler/grpc/handler.go [hunk 8]
```diff
@@ -96,63 +96,63 @@ func (s *server) Locations(ctx context.Context, req *payload.Backup_Locations_Re
...
+	for _, vector := range vectors.Vectors {
+		var m *model.Vector
+		m, err = toModelVector(vector)
 		if err != nil {
 			log.Errorf("[RegisterMulti]\tunknown error\t%+v", err)
 			if span != nil {
 				span.SetStatus(trace.StatusCodeInternal(err.Error()))
 			}
-			return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API uuids %s's could not convert vector to meta_vector", meta.GetUuid()), err, info.Get())
+			return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API uuids %s's could not convert vector to vector_vector", vector.GetUuid()), err, info.Get())
```
Comment (kmrmt) [inline pkg/manager/backup/mysql/handler/grpc/handler.go:144]: same reason(strange error message)

File: pkg/manager/backup/mysql/handler/grpc/handler.go [hunk 9]
```diff
@@ -96,63 +96,63 @@ func (s *server) Locations(ctx context.Context, req *payload.Backup_Locations_Re
...
 
-	err = s.mysql.SetMetas(ctx, ms...)
+	err = s.mysql.SetVectors(ctx, ms...)
 	if err != nil {
 		log.Errorf("[RegisterMulti]\tunknown error\t%+v", err)
 		if span != nil {
 			span.SetStatus(trace.StatusCodeInternal(err.Error()))
 		}
-		return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API failed to backup metadatas %#v", ms), err, info.Get())
+		return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API failed to backup vectordatas %#v", ms), err, info.Get())
```
Comment (kmrmt) [inline pkg/manager/backup/mysql/handler/grpc/handler.go:155]: same reason(data is unnecessay)

**Review by kmrmt — CHANGES_REQUESTED** [review]

File: internal/db/rdb/mysql/mysql.go (hunk 1 above)
Comment (kmrmt) [reply internal/db/rdb/mysql/mysql.go:177]: `vector data's vector` is too much. I think `vector` is enough.

File: internal/db/rdb/mysql/mysql.go [hunk 10]
```diff
@@ -251,23 +250,23 @@ func (m *mySQLClient) SetMeta(ctx context.Context, mv MetaVector) error {
...
 	defer tx.RollbackUnlessCommitted()
 
-	err = validateMeta(mv)
+	err = validateVector(mv)
 	if err != nil {
 		return err
 	}
 
-	_, err = tx.InsertBySql("INSERT INTO meta_vector(uuid, vector, meta) VALUES (?, ?, ?) ON DUPLICATE KEY UPDATE vector = ?, meta = ?",
+	_, err = tx.InsertBySql("INSERT INTO "+vectorTableName+"(uuid, vector, meta) VALUES (?, ?, ?) ON DUPLICATE KEY UPDATE vector = ?, meta = ?",
```
Comment (rinx) [inline internal/db/rdb/mysql/mysql.go:258]:
```suggestion
	_, err = tx.InsertBySql("INSERT INTO "+vectorTableName+"(uuid, vector) VALUES (?, ?) ON DUPLICATE KEY UPDATE vector = ?",
```
and please remove `mv.GetVector()` on L261 and L262.

File: internal/db/rdb/mysql/mysql.go [hunk 11]
```diff
@@ -305,25 +304,25 @@ func (m *mySQLClient) SetMetas(ctx context.Context, metas ...MetaVector) error {
...
 
 	for _, meta := range metas {
-		err = validateMeta(meta)
+		err = validateVector(meta)
 		if err != nil {
 			return err
 		}
 
-		_, err = tx.InsertBySql("INSERT INTO meta_vector(uuid, vector, meta) VALUES (?, ?, ?) ON DUPLICATE KEY UPDATE vector = ?, meta = ?",
+		_, err = tx.InsertBySql("INSERT INTO "+vectorTableName+"(uuid, vector, meta) VALUES (?, ?, ?) ON DUPLICATE KEY UPDATE vector = ?, meta = ?",
```
Comment (rinx) [inline internal/db/rdb/mysql/mysql.go:312]: same
```suggestion
		_, err = tx.InsertBySql("INSERT INTO "+vectorTableName+"(uuid, vector) VALUES (?, ?) ON DUPLICATE KEY UPDATE vector = ?",
```
also remove L315 and L316

File: internal/db/rdb/mysql/mysql.go [hunk 12]
```diff
@@ -251,23 +250,23 @@ func (m *mySQLClient) SetMeta(ctx context.Context, mv MetaVector) error {
...
 
-	_, err = tx.InsertBySql("INSERT INTO meta_vector(uuid, vector, meta) VALUES (?, ?, ?) ON DUPLICATE KEY UPDATE vector = ?, meta = ?",
+	_, err = tx.InsertBySql("INSERT INTO "+vectorTableName+"(uuid, vector, meta) VALUES (?, ?, ?) ON DUPLICATE KEY UPDATE vector = ?, meta = ?",
 		mv.GetUUID(),
 		mv.GetVector(),
-		mv.GetMeta(),
 		mv.GetVector(),
-		mv.GetMeta()).ExecContext(ctx)
+		mv.GetVector(),
+		mv.GetVector()).ExecContext(ctx)
```
Comment (kmrmt) [inline internal/db/rdb/mysql/mysql.go:263]: It is necessary to delete meta column.

File: internal/db/rdb/mysql/mysql.go [hunk 13]
```diff
@@ -305,25 +304,25 @@ func (m *mySQLClient) SetMetas(ctx context.Context, metas ...MetaVector) error {
...
 
-		_, err = tx.InsertBySql("INSERT INTO meta_vector(uuid, vector, meta) VALUES (?, ?, ?) ON DUPLICATE KEY UPDATE vector = ?, meta = ?",
+		_, err = tx.InsertBySql("INSERT INTO "+vectorTableName+"(uuid, vector, meta) VALUES (?, ?, ?) ON DUPLICATE KEY UPDATE vector = ?, meta = ?",
 			meta.GetUUID(),
 			meta.GetVector(),
-			meta.GetMeta(),
 			meta.GetVector(),
-			meta.GetMeta()).ExecContext(ctx)
+			meta.GetVector(),
+			meta.GetVector()).ExecContext(ctx)
```
Comment (kmrmt) [inline internal/db/rdb/mysql/mysql.go:317]: same reason(delete meta)

File: internal/db/rdb/mysql/set.go [hunk 14]
```diff
@@ -19,10 +19,10 @@ package mysql
 import "context"
 
 type Setter interface {
-	SetMeta(ctx context.Context, meta MetaVector) error
-	SetMetas(ctx context.Context, metas ...MetaVector) error
-	DeleteMeta(ctx context.Context, uuid string) error
-	DeleteMetas(ctx context.Context, uuids ...string) error
+	SetVector(ctx context.Context, meta Vector) error
+	SetVectors(ctx context.Context, metas ...Vector) error
```
Comment (rinx) [inline internal/db/rdb/mysql/set.go:23]:
```suggestion
	SetVector(ctx context.Context, vector Vector) error
	SetVectors(ctx context.Context, vectors ...Vector) error
```

File: internal/db/rdb/mysql/mysql.go (hunk 1 above)
Comment (kpango) [reply internal/db/rdb/mysql/mysql.go:177]: Could you please use github suggestion for each comment ?

File: pkg/agent/core/ngt/config/config_test.go [hunk 15]
```diff
@@ -98,6 +98,7 @@ func TestNewConfig(t *testing.T) {
 			if err := test.checkFunc(test.want, gotCfg, err); err != nil {
 				tt.Errorf("error = %v", err)
 			}
+
```
Comment (rinx) [inline pkg/agent/core/ngt/config/config_test.go:101]: it doesn't affect the codes but... why these blank lines appeared?

File: internal/db/rdb/mysql/mysql.go (hunk 1 above)
Comment (kmrmt) [inline internal/db/rdb/mysql/mysql.go]:
```suggestion
// GetVector gets the vector data and podIPs which have index of vector.
```

File: internal/db/rdb/mysql/mysql.go (hunk 10 above)
Comment (kmrmt) [inline internal/db/rdb/mysql/mysql.go]:
```suggestion
	_, err = tx.InsertBySql("INSERT INTO "+vectorTableName+"(uuid, vector) VALUES (?, ?) ON DUPLICATE KEY UPDATE vector = ?",
```

File: pkg/agent/core/ngt/config/config_test.go (hunk 15 above)
Comment (kpango) [reply pkg/agent/core/ngt/config/config_test.go:101]: I don't know

File: internal/db/rdb/mysql/mysql.go (hunk 11 above)
Comment (kmrmt) [inline internal/db/rdb/mysql/mysql.go]:
```suggestion
		_, err = tx.InsertBySql("INSERT INTO "+vectorTableName+"(uuid, vector) VALUES (?, ?) ON DUPLICATE KEY UPDATE vector = ?",
```

File: pkg/manager/backup/cassandra/handler/grpc/handler.go (hunk 5 above)
Comment (kmrmt) [inline pkg/manager/backup/cassandra/handler/grpc/handler.go]:
```suggestion
			return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API uuids %s's could not convert vector", vector.GetUuid()), err, info.Get())
```

File: pkg/manager/backup/cassandra/handler/grpc/handler.go (hunk 6 above)
Comment (kmrmt) [inline pkg/manager/backup/cassandra/handler/grpc/handler.go]:
```suggestion
		return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API failed to backup vectors %#v", ms), err, info.Get())
```

File: pkg/manager/backup/cassandra/handler/grpc/handler.go (hunk 4 above)
Comment (kmrmt) [inline pkg/manager/backup/cassandra/handler/grpc/handler.go]:
```suggestion
		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's failed to backup vector", uuid), err, info.Get())
```

File: pkg/manager/backup/cassandra/handler/grpc/handler.go (hunk 3 above)
Comment (kmrmt) [inline pkg/manager/backup/cassandra/handler/grpc/handler.go]:
```suggestion
		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's could not convert vector", uuid), err, info.Get())
```

File: pkg/manager/backup/mysql/handler/grpc/handler.go (hunk 8 above)
Comment (kmrmt) [inline pkg/manager/backup/mysql/handler/grpc/handler.go]:
```suggestion
			return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API uuids %s's could not convert vector", vector.GetUuid()), err, info.Get())
```

File: pkg/manager/backup/mysql/handler/grpc/handler.go (hunk 9 above)
Comment (kmrmt) [inline pkg/manager/backup/mysql/handler/grpc/handler.go]:
```suggestion
		return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API failed to backup vectors %#v", ms), err, info.Get())
```

**Comment by kpango** [comment]
@rinx @kmrmt fixed

File: pkg/manager/backup/mysql/handler/grpc/handler.go (hunk 7 above)
Comment (kmrmt) [inline pkg/manager/backup/mysql/handler/grpc/handler.go]:
```suggestion
		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's failed to backup vector", uuid), err, info.Get())
```

File: pkg/manager/backup/mysql/handler/grpc/handler.go (hunk 2 above)
Comment (kmrmt) [inline pkg/manager/backup/mysql/handler/grpc/handler.go]:
```suggestion
		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's could not convert vector", uuid), err, info.Get())
```

**Review by kmrmt — CHANGES_REQUESTED** [review]

File: pkg/manager/backup/cassandra/handler/grpc/handler.go [hunk 16]
```diff
@@ -105,63 +105,63 @@ func (s *server) Locations(ctx context.Context, req *payload.Backup_Locations_Re
...
-	m, err := toModelMetaVector(meta)
+	uuid := vector.GetUuid()
+	m, err := toModelVector(vector)
 	if err != nil {
 		log.Errorf("[Register]\tunknown error\t%+v", err)
 		if span != nil {
 			span.SetStatus(trace.StatusCodeInternal(err.Error()))
 		}
-		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's could not convert vector to meta_vector", uuid), err, info.Get())
+		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's could not convert vector to vector", uuid), err, info.Get())
```
Comment (kmrmt) [inline pkg/manager/backup/cassandra/handler/grpc/handler.go:122]:
```suggestion
		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's could not convert vector to backup format", uuid), err, info.Get())
```

File: pkg/manager/backup/cassandra/handler/grpc/handler.go [hunk 17]
```diff
@@ -105,63 +105,63 @@ func (s *server) Locations(ctx context.Context, req *payload.Backup_Locations_Re
...
+	for _, vector := range vectors.Vectors {
+		var m *model.Vector
+		m, err = toModelVector(vector)
 		if err != nil {
 			log.Errorf("[RegisterMulti]\tunknown error\t%+v", err)
 			if span != nil {
 				span.SetStatus(trace.StatusCodeInternal(err.Error()))
 			}
-			return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API uuids %s's could not convert vector to meta_vector", meta.GetUuid()), err, info.Get())
+			return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API uuids %s's could not convert vector to vector", vector.GetUuid()), err, info.Get())
```
Comment (kmrmt) [inline pkg/manager/backup/cassandra/handler/grpc/handler.go:153]:
```suggestion
			return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API uuids %s's could not convert vector to backup format", vector.GetUuid()), err, info.Get())
```

File: pkg/manager/backup/mysql/handler/grpc/handler.go [hunk 18]
```diff
@@ -96,63 +96,63 @@ func (s *server) Locations(ctx context.Context, req *payload.Backup_Locations_Re
...
-	m, err := toModelMetaVector(meta)
+	uuid := vector.GetUuid()
+	m, err := toModelVector(vector)
 	if err != nil {
 		log.Errorf("[Register]\tunknown error\t%+v", err)
 		if span != nil {
 			span.SetStatus(trace.StatusCodeInternal(err.Error()))
 		}
-		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's could not convert vector to meta_vector", uuid), err, info.Get())
+		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's could not convert vector to vector", uuid), err, info.Get())
```
Comment (kmrmt) [inline pkg/manager/backup/mysql/handler/grpc/handler.go:113]:
```suggestion
		return nil, status.WrapWithInternal(fmt.Sprintf("Register API uuid %s's could not convert vector to backup format", uuid), err, info.Get())
```

File: pkg/manager/backup/mysql/handler/grpc/handler.go [hunk 19]
```diff
@@ -96,63 +96,63 @@ func (s *server) Locations(ctx context.Context, req *payload.Backup_Locations_Re
...
+	for _, vector := range vectors.Vectors {
+		var m *model.Vector
+		m, err = toModelVector(vector)
 		if err != nil {
 			log.Errorf("[RegisterMulti]\tunknown error\t%+v", err)
 			if span != nil {
 				span.SetStatus(trace.StatusCodeInternal(err.Error()))
 			}
-			return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API uuids %s's could not convert vector to meta_vector", meta.GetUuid()), err, info.Get())
+			return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API uuids %s's could not convert vector to vector", vector.GetUuid()), err, info.Get())
```
Comment (kmrmt) [inline pkg/manager/backup/mysql/handler/grpc/handler.go:144]:
```suggestion
			return nil, status.WrapWithInternal(fmt.Sprintf("RegisterMulti API uuids %s's could not convert vector to backup format", vector.GetUuid()), err, info.Get())
```

**Review by kmrmt — CHANGES_REQUESTED** [review]
The message is example.

**Comment by kpango** [comment]
@kmrmt fixed

**Review by kmrmt — APPROVED** [review]
LGTM

**Comment by kpango** [comment]
@rinx could you please re-review this PR?

**Comment by rinx** [comment]
overall it seems to be ok.

**Comment by kpango** [comment]
@rinx @kmrmt  thank you for review, I'll merge this PR into #740, and I'm working on fixing hack package for V1 after that I'll request review of #740

**Pull request merged**